moodlecli/utils.py

```diff
@@ -193,3 +194,19 @@ def setup_duplicate_course(
     coursedata[CSV_COURSE_ENROLMENT_KEY] = enrolment_key
 
     return coursedata
+
+
+def inject_uuids(uuid_data, user_data):
+    uuid_map = {}
+    # Make a hash of all user_ids to uuids
+    for item in uuid_data:
+        uuid_map[int(item['user_id'])] = item['user_uuid']
+
+    # Add uuids to user data
+    for user in user_data:
+        user_id = int(user['id'])
+        if user_id in uuid_map.keys():
+            user['user_uuid'] = uuid_map[user_id]
+        else:
+            user['user_uuid'] = str(UUID(int=random.getrandbits(128)))
```

Comment:
Let's not put this business logic of a random UUID here. I would either not add it if we don't have one, or make it `None` / `null`.

Reply:
Yesterday we were thinking that if a user is enrolled in a course but hasn't interacted with the course yet we'd hit this case where the uuid has no mapping for a user - then, in order to prevent joins on a none value in the ETL - we figured it'd be better to have a fake uuid rather than a None value. 

The only reason this has the `random.getrandombits` part is so that the uuid is deterministic for testing... I realize this is coding for tests... but added it anyway bc I felt like determinism was nice, regardless. 

I think of these choices tho ^ I think making the value None would be better than leaving field out to prevent keying errors.